tests/test_media.py

```diff
@@ -2,90 +2,119 @@
-        assert standard_data[4][4] == 4 * 4
+
+    def test_set_image_dimensions(self):
+        img = ViamImage(b"data", CameraMimeType.JPEG)
+        with pytest.raises(AttributeError):
+            img.width
+        with pytest.raises(AttributeError):
+            img.height
+        img.width = 100
+        img.height = 100
+        assert img.width == 100
+        assert img.height == 100
 
 
 class TestNamedImage:
     def test_name(self):
         name = "img"
-        img = NamedImage(name, b"123", CameraMimeType.UNSUPPORTED)
+        img = NamedImage(name, b"123", CameraMimeType.JPEG)
         assert img.name == name
+
+
+def test_viam_to_pil_image():
+    i = Image.new("RGBA", (100, 100), "#AABBCCDD")
+    b = BytesIO()
+    i.save(b, "PNG")
+    img = ViamImage(b.getvalue(), CameraMimeType.JPEG)
+    pil_img = viam_to_pil_image(img)
+    assert pil_img.tobytes() == i.tobytes()
+
+
+def test_bytes_to_depth_array():
+    with open(f"{os.path.dirname(__file__)}/../data/fakeDM.vnd.viam.dep", "rb") as depth_map:
+        img = ViamImage(depth_map.read(), CameraMimeType.VIAM_RAW_DEPTH)
+    assert isinstance(img, ViamImage)
+    standard_data = bytes_to_depth_array(img)
+    assert len(standard_data) == 10
+    assert len(standard_data[0]) == 20
+    data_arr = array("H", img.data[24:])
+    data_arr.byteswap()
+    assert len(data_arr) == 200
+    assert standard_data[0][0] == data_arr[0]
+    assert standard_data[-1][3] == data_arr[183]
+    assert standard_data[-1][3] == 9 * 3
+    assert standard_data[4][4] == 4 * 4
+
+    img2 = ViamImage(b"data", CameraMimeType.PCD)
+    with pytest.raises(NotSupportedError):
+        bytes_to_depth_array(img2)
+
+
+def test_determine_image_dimensions():
+    i = Image.new("RGBA", (100, 100), "#AABBCCDD")
+    b = BytesIO()
+    i.save(b, "PNG")
+    img = ViamImage(b.getvalue(), CameraMimeType.PNG)
+    with pytest.raises(AttributeError):
+        img.width
+    with pytest.raises(AttributeError):
+        img.height
+    determine_image_dimensions(img)
+    assert img.width == 100
+    assert img.height == 100
+
+    i2 = Image.new("RGBA", (100, 100), "#AABBCCDD")
+    b2 = BytesIO()
+    i2.save(b2, "VIAM_RGBA")
+    img2 = ViamImage(b.getvalue(), CameraMimeType.VIAM_RGBA)
+    with pytest.raises(AttributeError):
+        img2.width
+    with pytest.raises(AttributeError):
+        img2.height
+    determine_image_dimensions(img2)
+    assert img2.width == 100
+    assert img2.height == 100
+
+    i3 = Image.new("RGB", (100, 100), "#AABBCCDD")
+    b3 = BytesIO()
+    i3.save(b3, "JPEG")
+    img3 = ViamImage(b.getvalue(), CameraMimeType.JPEG)
+    with pytest.raises(AttributeError):
+        img3.width
+    with pytest.raises(AttributeError):
+        img3.height
+    determine_image_dimensions(img3)
+    assert img3.width == 100
+    assert img3.height == 100
+
+    img4 = ViamImage(b"data", CameraMimeType.PCD)
+    with pytest.raises(NotSupportedError):
+        determine_image_dimensions(img4)
```

Comment:
You should at MimeType tests to make sure that when you pass in mime types we support and ones we don't, that everything still works (e.g. that `MimeType.from_string("SOME_RANDOM_STRING")` doesn't error)

Reply:
Would there be a `ViamImage` with an unsupported mime type? We purposely removed `CameraMimeType.UNSUPPORTED`, and `ViamImage` takes a `CameraMimeType` object.

Reply:
All good